core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -146,6 +146,9 @@ class GroupCoordinator(val brokerId: Int,
       if (!group.is(Empty) && (!group.protocolType.contains(protocolType) || !group.supportsProtocols(protocols.map(_._1).toSet))) {
         // if the new member does not support the group protocol, reject it
         responseCallback(joinError(memberId, Errors.INCONSISTENT_GROUP_PROTOCOL))
+      } else if (group.is(Empty) && protocols.isEmpty ) {
```

Comment:
Maybe we should also check whether protocolType is non-empty?

Reply:
added a non-empty check on protocolType